huaweicloud/services/modelarts/resource_huaweicloud_modelarts_service.go

```diff
@@ -0,0 +1,1007 @@
+				Optional: true,
+				Computed: true,
+			},
+		},
+	}
+	return &sc
+}
+
+func modelartsServiceAdditionalPropertySmnNotificationSchema() *schema.Resource {
+	sc := schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"topic_urn": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				Description: `URN of an SMN topic.`,
+			},
+			"events": {
+				Type:        schema.TypeList,
+				Elem:        &schema.Schema{Type: schema.TypeInt},
+				Optional:    true,
+				Computed:    true,
+				Description: `Event ID.`,
+			},
+		},
+	}
+	return &sc
+}
+
+func modelartsServiceAdditionalPropertyLogReportChannelSchema() *schema.Resource {
+	sc := schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"type": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				Description: `The type of log report channel. The valid value is **LTS**.`,
+			},
+		},
+	}
+	return &sc
+}
+
+func resourceModelartsServiceCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+
+	// createService: create a ModelArts service.
+	var (
+		createServiceHttpUrl = "v1/{project_id}/services"
+		createServiceProduct = "modelarts"
+	)
+	createServiceClient, err := cfg.NewServiceClient(createServiceProduct, region)
+	if err != nil {
+		return diag.Errorf("error creating ModelArts Client: %s", err)
+	}
+
+	createServicePath := createServiceClient.Endpoint + createServiceHttpUrl
+	createServicePath = strings.ReplaceAll(createServicePath, "{project_id}", createServiceClient.ProjectID)
+
+	createServiceOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		OkCodes: []int{
+			200,
+		},
+		MoreHeaders: map[string]string{"Content-Type": "application/json;charset=UTF-8"},
+	}
+
+	createServiceOpt.JSONBody = utils.RemoveNil(buildCreateServiceBodyParams(d))
+	createServiceResp, err := createServiceClient.Request("POST", createServicePath, &createServiceOpt)
+	if err != nil {
+		return diag.Errorf("error creating ModelartsService: %s", err)
+	}
+
+	createServiceRespBody, err := utils.FlattenResponse(createServiceResp)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	id, err := jmespath.Search("service_id", createServiceRespBody)
+	if err != nil {
+		return diag.Errorf("error creating ModelartsService: ID is not found in API response")
+	}
+	d.SetId(id.(string))
+
+	err = serviceWaitingForStateCompleted(ctx, d, meta, d.Timeout(schema.TimeoutCreate))
+	if err != nil {
+		return diag.Errorf("error waiting for the Create of ModelartsService (%s) to complete: %s", d.Id(), err)
+	}
+	return resourceModelartsServiceRead(ctx, d, meta)
+}
+
+func buildCreateServiceBodyParams(d *schema.ResourceData) map[string]interface{} {
+	bodyParams := map[string]interface{}{
+		"workspace_id":          utils.ValueIngoreEmpty(d.Get("workspace_id")),
+		"service_name":          utils.ValueIngoreEmpty(d.Get("name")),
```

Comment:
the name, infer_type, config are required parameters, they would never be empty.

Reply:
done.

Reply:
but It's better to keep it independent from the schema. 
If a field in the schema is changed to be optional, we just need to change the schema.

Reply:
that will be the repsonsibility for update PR should have